catalog/rest_test.go

```diff
@@ -114,6 +114,39 @@ func (r *RestCatalogSuite) TestToken200() {
 	r.Equal(r.configVals.Get("warehouse"), "s3://some-bucket")
 }
 
+func (r *RestCatalogSuite) TestLoadRegisteredCatalog() {
+	r.mux.HandleFunc("/v1/oauth/tokens", func(w http.ResponseWriter, req *http.Request) {
```

Comment:
FYI `/v1/oauth/tokens` is deprecated 
https://github.com/apache/polaris/blob/deb1c879b05c691cf23cdc4ab0ca2ec20729926a/spec/rest-catalog-open-api.yaml#L171-L182

Reply:
Currently in the Rest catalog it's used as a fallback if there's no auth uri set. If this is being deprecated, what should the workflow be if the authuri isn't set and there's no credential or token set?

Reply:
> If this is being deprecated, what should the workflow be if the authuri isn't set and there's no credential or token set? Should it just error instead of falling back to that?

I checked the spec and both the java and python implementation, i dont think there's any examples of this yet. 
I would assume it should just error since 
```
        All clients are encouraged to explicitly set the configuration property `oauth2-server-uri`
        to the correct OAuth endpoint.
```

The overall discussion for this is in https://github.com/apache/iceberg/issues/10537

Reply:
I think its fine to leave this for now, i just wanted to mention that its deprecated and will be removed in 2.0 release